api/v1alpha1/externaldns_types.go

```diff
@@ -355,15 +357,9 @@ type ExternalDNSInfobloxProviderOptions struct {
 	WAPIVersion string `json:"wapiVersion"`
 }
 
-// NamespacedSecretReference contains enough information to let you locate the
-// desired secret in a specific namespace.
-type NamespacedSecretReference struct {
-	// Namespace is the namespace of the secret.
-	//
-	// +kubebuilder:validation:Required
-	// +required
-	Namespace string `json:"namespace"`
-
+// SecretReference contains the information to let you locate the desired secret.
+// Secret is expected to be in the operator namespace.
```

Comment:
Secret is required to be in the operator namespace

Reply:
This is weird actually.  You have a cluster scoped resource that refers to a secret in a particular namespace that can change depending on where the operator is installed?

Can you ever have more than one operator installed?  Do you know what namespace this is?  operator or operand?

Reply:
Yes, the namespace can change depending on where the operator is installed. The cluster admin will have to give this namespace to OLM at the installation. The secret will have to be there before the CR is created.

> Can you ever have more than one operator installed?

Technically, it's possible. But the idea is 1 operator for the whole cluster.

> Do you know what namespace this is? operator or operand?

The secret which was created by the cluster admin in the operator namespace will be copied into the operand namespace by the operator.

Also, an important note: if the secret is not referenced in the CR, the operator will request one from the cloud credentials operator (if the underlying platform is OpenShift).

Reply:
> Yes, the namespace can change depending on where the operator is installed. The cluster admin will have to give this namespace to OLM at the installation. The secret will have to be there before the CR is created.

This level of flexibility seems unnecessary and undesireable.  Why should the namespace be configurable?

Reply:
> Why should the namespace be configurable?

Technically the operator (its manifests) is agnostic to the namespace in which it will be installed. But I believe this is an OLM requirement (`OperatorGroup.spec.targetNamespaces`).